docker-app/qfieldcloud/core/tests/test_team.py

```diff
@@ -0,0 +1,410 @@
+            f"/api/v1/organizations/{self.o1.username}/teams/{self.t1.teamname}/",
+            {"username": "newteamname"},
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        updated_team_username = Team.format_team_name(self.o1.username, "newteamname")
+
+        self.t1.refresh_from_db()
+
+        self.assertEqual(self.t1.username, updated_team_username)
+        self.assertEqual(Team.objects.filter(username=updated_team_username).count(), 1)
+
+    def test_admin_user_cannot_update_team_with_empty_payload(self):
+        """Test that an admin can update team without a payload."""
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token2.key)
+
+        old_team_name = self.t1.teamname
+
+        response = self.client.put(
+            f"/api/v1/organizations/{self.o1.username}/teams/{self.t1.teamname}/",
+            {},
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        self.t1.refresh_from_db()
+
+        self.assertEqual(self.t1.teamname, old_team_name)
+
+    def test_unauthorized_user_cannot_delete_team(self):
+        """Ensure unauthorized users cannot delete a team."""
+        self.assertEqual(Team.objects.filter(username=self.t1.username).count(), 1)
+
+        response = self.client.delete(
+            f"/api/v1/organizations/{self.o1.username}/teams/{self.t1.teamname}/"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertEqual(Team.objects.filter(username=self.t1.username).count(), 1)
+
+    def test_non_admin_user_cannot_delete_team(self):
+        """Ensure non-admin members cannot delete a team."""
+        self.assertEqual(Team.objects.filter(username=self.t1.username).count(), 1)
+
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token3.key)
+
+        response = self.client.delete(
+            f"/api/v1/organizations/{self.o1.username}/teams/{self.t1.teamname}/"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertEqual(Team.objects.filter(username=self.t1).count(), 1)
+
+    def test_unauthorized_user_cannot_list_teams(self):
+        """Test that an unauthorized user cannot list teams."""
+        response = self.client.get(f"/api/v1/organizations/{self.o1.username}/teams/")
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_admin_user_can_delete_team(self):
+        self.assertEqual(Team.objects.filter(username=self.t1.username).count(), 1)
+
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token1.key)
+
+        response = self.client.delete(
+            f"/api/v1/organizations/{self.o1.username}/teams/{self.t1.teamname}/"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(Team.objects.filter(username=self.t1.username).count(), 0)
```

Comment:
```suggestion
    def test_admin_user_can_delete_team(self):
        self.assertEqual(Team.objects.filter(username=self.t1.username).count(), 1)

        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token1.key)

        response = self.client.delete(
            f"/api/v1/organizations/{self.o1.username}/teams/{self.t1.teamname}/"
        )

        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
        self.assertEqual(Team.objects.filter(username=self.t1.username).count(), 0)
        
    def test_unauthorized_user_cannot_list_teams(self):
        """Test that an unauthorized user cannot list teams."""
        response = self.client.get(f"/api/v1/organizations/{self.o1.username}/teams/")

        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
```

Just swapping the order of the tests, all "deletes" together, all "lists" together. Makes it easier to read.

Reply:
indeed!

Reply:
done